src/transformers/modeling_gguf_pytorch_utils.py

```diff
@@ -55,6 +55,183 @@
 GGUF_SUPPORTED_ARCHITECTURES = list(GGUF_TO_TRANSFORMERS_MAPPING["tensors"].keys())
 
 
+class TensorProcessor:
+    def process(self, weights, name, **kwargs):
+        return {"weights": weights, "name": name, "metadata": {}}
+
+
+class LlamaTensorProcessor(TensorProcessor):
+    def process(self, weights, name, **kwargs):
+        if ".attn_k." in name or ".attn_q." in name:
+            config = kwargs.get("config", {})
+            num_heads = config.get("num_attention_heads")
+            num_kv_heads = config.get("num_key_value_heads")
```

Comment:
Can we initialize `TensorProcessor` class with config? So that we can focus on weights processing without considering config inputs. Otherwise we need to parse config when processing each tensor.

Reply:
yes, sure, fixed